contracts/SpokePoolV3Periphery.sol

```diff
@@ -53,153 +84,153 @@ contract SpokePoolPeripheryProxy is Lockable, MultiCaller {
-        try swapToken.permit(msg.sender, address(this), swapTokenAmount, deadline, v, r, s) {} catch {}
-        _callSwapAndBridge(
-            IERC20(address(swapToken)),
-            acrossInputToken,
-            exchange,
-            routerCalldata,
-            swapTokenAmount,
-            minExpectedInputTokenAmount,
-            depositData
+        try
+            IERC20Permit(address(swapData.swapToken)).permit(
+                msg.sender,
+                address(this),
+                swapData.swapTokenAmount,
+                deadline,
+                v,
+                r,
+                s
+            )
+        {} catch {}
+        _callSwapAndBridge(acrossInputToken, swapData, depositData);
+    }
+
+    /**
+     * @notice Uses permit2 to transfer tokens from a user before swapping a token on this chain via specified router and submitting an Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev This function assumes the caller has properly set an allowance for the permit2 contract on this network.
+     * @dev This function assumes that the amount of token to be swapped is equal to the amount of the token to be received from permit2.
+     * @param acrossInputToken Address of the token that will be bridged via Across as the inputToken.
+     * @param depositor The owner of the permit2 signature and depositor for the Across spoke pool.
+     * @param swapData Specifies the params we need to perform a swap on a generic exchange.
+     * @param depositData Specifies the Across deposit params we'll send after the swap.
+     * @param permit The permit data signed over by the owner.
+     * @param transferDetails The spender's requested transfer details for the permitted token.
+     * @param signature The permit2 signature to verify against the deposit data.
+     */
+    function swapAndBridgeWithPermit2(
+        IERC20 acrossInputToken,
+        address depositor,
+        SpokePoolV3Periphery.SwapData calldata swapData,
+        SpokePoolV3Periphery.DepositData calldata depositData,
+        IPermit2.PermitTransferFrom calldata permit,
+        IPermit2.SignatureTransferDetails calldata transferDetails,
+        bytes calldata signature
+    ) external nonReentrant {
+        bytes32 witness = keccak256(
+            abi.encode(
+                ACROSS_DEPOSIT_DATA_TYPEHASH,
+                depositData.outputToken,
+                depositData.outputAmount,
+                depositData.depositor,
+                depositData.recipient,
+                depositData.destinationChainId,
+                depositData.exclusiveRelayer,
+                depositData.quoteTimestamp,
+                depositData.fillDeadline,
+                depositData.exclusivityParameter,
+                depositData.message
+            )
+        );
+        permit2.permitWitnessTransferFrom(
+            permit,
+            transferDetails,
```

Comment:
Shouldn't we be enforcing that the transferDetails.to == address(this)? We only want to pull funds into this contract, right?

Reply:
We can, but if somebody did this, I think it means that they just passed in some random permit signature. The function would revert eventually, since the periphery would have no funds to swap or deposit.